### ISSUE TITLE
polychord interface?

### ISSUE THREAD

**Issue by jtlz2** [JohannesBuchner/PyMultiNest#52]
@JohannesBuchner Any chance of a python interface to polychord, the successor to MultiNest? :) http://arxiv.org/abs/1502.01856

**Comment by JohannesBuchner** [comment]
Patches welcome ;)

**Comment by jtlz2** [comment]
How far would I get with f2py? :|

**Comment by JohannesBuchner** [comment]
I think you can just try to take the cwrapper module in multinest https://github.com/JohannesBuchner/MultiNest/blob/master/src/cwrapper.f90 and adapt it to the fortran code in polychord. The entry point is NestedSampling as used in main.f https://github.com/JohannesBuchner/PolyChord/blob/master/src/main.F90

**Comment by jtlz2** [comment]
v1.1 has a fortran_cpp_wrapper.f90 ?

**Comment by jtlz2** [comment]
@JohannesBuchner A pypolychord now exists - perhaps this would be a good place to host it? @vhaasteren has written one...

**Comment by vhaasteren** [comment]
Yes, I wrote one back in February. Actually, I just used PyMultiNest as an example for it. But I never got it to work with the newest version of PolyChord. My own git repo contains polychord as well (many many edits to it were necessary, for it to work), so that's why I did not host it here.

**Comment by JohannesBuchner** [comment]
@vhaasteren Would you like me to adopt it into PyMultiNest or would you prefer to keep control over it?

**Comment by vhaasteren** [comment]
If you want to adopt it into PyMultiNest, then by all means, go ahead!

**Comment by JohannesBuchner** [comment]
Please make your modified polychord repo public too, and link both here. (The polychord library explicitly allows distribution and modification, for academic/non-commercial use).

**Comment by vhaasteren** [comment]
You are right. Fine, I have made it publicly available through: https://github.com/vhaasteren/pypolychord

The README.md contains instructions on how to initially make it work. I hope it will be useful. I might spend some more time on it in the future, but just not right now. Thank you for your interest!

**Comment by JohannesBuchner** [comment]
I hate Fortran ....

**Comment by JohannesBuchner** [comment]
Well, there is a prototype in my repos now.

Get (modified) PolyChord 1.4 from here:
   https://github.com/JohannesBuchner/PolyChord/
which includes those likelihood call modifications.
Perhaps we could put a intermediate call in between that substitutes for the extended call (with ndim and nderived).
Alternatively, Perhaps there is a way to use likelihoods/fortran_cpp_wrapper.f90, which works with fortran's iso_c_binding (like we did for newer versions of PyMultiNest).

In any case, here is how to use it in Python:
https://github.com/JohannesBuchner/PyMultiNest/blob/master/pypolychord_demo.py
The solution I chose writes an ini file and thus the programming interface does not need to be updated and should gracefully downgrade.

Problems:
  - for the eggbox problem, it hangs sometimes, not finding a higher point. This seems to occur when there are too many clusters or when they are towards the edges (?).
  - When setting nlive != 500 there seem to be problems sometimes.
